SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -10,50 +14,63 @@ namespace SkietbaanBE.Controllers
     public class UserController : Controller
     {
         private ModelsContext _context;
-        public UserController(ModelsContext db) {
+        public UserController(ModelsContext db)
+        {
             _context = db;
         }
-
-        //public UserController() { }
-
         // GET: api/User
         [HttpGet]
-        public IEnumerable<User> Get() {
-            var users = _context.Users.ToArray<User>();
-            return users;
+        public IEnumerable<User> Get()
+        {
+           return _context.Users.ToArray<User>();
         }
-
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
-            User user = _context.Users.Find(id);
-            return user;
-        }
-
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
+        public async Task<User> Get(int id)
+        {
+            return await _context.Users.FindAsync(id);
         }
-
         // POST: api/User
-        [HttpPost]
-        public virtual void Post([FromBody] User user) {
-            _context.Add(user);
-            _context.SaveChanges();
+        [HttpPost("{id}")]
+        public async Task<HttpResponseMessage> Post(int id,[FromBody] User user)
+        {
+            if (ModelState.IsValid)
+            {
+                //get user with the specified ID from database
+                User dbUser = await _context.Users.FindAsync(id);
+                //user not found
+                if(dbUser == null)
+                {
+                    return new HttpResponseMessage(HttpStatusCode.NotFound);
```

Comment:
You can make the return type of the function `Task<IActionResult>` and just `return NotFound();` here

Reply:
resolving now, will do a new PR in a sec